src/IECoreMaya/SceneShapeSubSceneOverride.cpp

```diff
@@ -79,6 +85,226 @@ using namespace MHWRender;
+				{
+						resultWritable[i] = in[triangulatedIndices[i]];
+				}
+			}
+		);
+
+		return result;
+	}
+
+	ConstV2fVectorDataPtr expandulateFacevarying( PrimitiveVariable::IndexedView<Imath::V2f> in )
+	{
+		const std::vector<int> &triangulatedIndices = m_triangulatedIndices->readable();
+
+		V2fVectorDataPtr result( new V2fVectorData() );
+		std::vector<Imath::V2f> &resultWritable = result->writable();
+		resultWritable.resize( triangulatedIndices.size() );
+
+		tbb::parallel_for(
+			tbb::blocked_range<int>( 0, triangulatedIndices.size() ),
+			[this, &in, &resultWritable](
+				const tbb::blocked_range<int> &range
+			)
+			{
+				for( int i = range.begin(); i != range.end(); ++i )
+				{
+					resultWritable[i] = in[m_mapToOldFacevarying[i]];
+				}
+			}
+		);
+
+		return result;
+	}
+
+	IECore::ConstIntVectorDataPtr indices()
+	{
+		return m_indices;
+	}
+
+	IECore::ConstIntVectorDataPtr triangulatedIndices()
+	{
+		return m_triangulatedIndices;
+	}
+
+	IECore::ConstIntVectorDataPtr wireframeIndices()
+	{
+		std::vector<int> &wireframeIndicesWritable = m_wireframeIndices->writable();
+
+		// We need to compute the wireframe only once per topology hash
+		if( wireframeIndicesWritable.size() )
+		{
+			return m_wireframeIndices;
+		}
+
+		const std::vector<int> &numVerticesPerFaceReadable = m_numVerticesPerFace->readable();
+
+		// Compute partial sum to make algorithm parallelisable.
+		// [4, 4, 4, 4] -> [4, 8, 12, 16]
+		std::vector<int> partialSum( numVerticesPerFaceReadable.size() );
+		std::partial_sum(numVerticesPerFaceReadable.begin(), numVerticesPerFaceReadable.end(), partialSum.begin(), std::plus<int>());
+
+		int totalNumEdgeIndices = partialSum[partialSum.size()-1] * 2;
+		wireframeIndicesWritable.resize( totalNumEdgeIndices );
+
+		const std::vector<int> &originalIndicesReadable = m_originalIndices->readable();
+
+		// Construct list of indices that can be used to render wireframes
+		tbb::parallel_for(
+			tbb::blocked_range<size_t>( 0, numVerticesPerFaceReadable.size() ), [this, &partialSum, &numVerticesPerFaceReadable, &originalIndicesReadable, &wireframeIndicesWritable]( const tbb::blocked_range<size_t> &r )
+			{
+				for( size_t i = r.begin(); i != r.end(); ++i )
+				{
+					int offset = i == 0 ? 0 : partialSum[i-1];
+					int numVerts = numVerticesPerFaceReadable[i];
+
+					for( int k = 0; k < numVerts; ++k )
+					{
+						int oldIndexA = originalIndicesReadable[offset + k];
+						int oldIndexB = originalIndicesReadable[offset + (k+1)%numVerts];
+
+						int insertion = (offset + k) * 2;
+
+						wireframeIndicesWritable[insertion] = m_oldToTriangulatedIndexMapping[ oldIndexA ];
+						wireframeIndicesWritable[insertion + 1] = m_oldToTriangulatedIndexMapping[ oldIndexB ];
+					}
+				}
+			}
+		);
+
+		return m_wireframeIndices;
+	}
+
+private :
+
+	size_t m_dataSize;
+
+	IECore::ConstIntVectorDataPtr m_numVerticesPerFace;
+
+	IECore::IntVectorDataPtr m_indices;
```

Comment:
I think I'd prefer `IECore[type]VectorDataPtrs` in the class and getting readable in each function or if you prefer keeping references to std::vector<int> for `m_oldToTriangulatedIndexMapping` & `m_mapToOldFacevarying` , but I don't see the need in this case.